### ISSUE TITLE
grpc_gateway_proto_library is not able to build

### ISSUE THREAD

**Issue by chaitanya9186** [pubref/rules_protobuf#86]
Build rule
````````
grpc_gateway_proto_library(
    name = "helloworld_proto_go",
    protos = ["helloworld.proto"],
    with_grpc = True,
)
`````````

helloworld.go
````````````
syntax = "proto3";

import "google/api/annotations.proto";

package helloworld;

// The greeting service definition.  The definition is repeated here
// rather than re-using examples/helloworld/proto/helloworld.proto so
// the other implementations / tests don't have to account for the
// annotations import.

service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {
    option (google.api.http) = {
      post: "/v1/helloworld/sayhello"
      body: "*"
    };
  }
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
`````````````````

Error
`````
ERROR: <blah blah>/pkg/BUILD:36:1: //<blah blah>/pkg:helloworld_proto_go: no such attribute 'with_grpc' in 'go_library' rule.
ERROR: error loading package '<blah blah>/pkg': Package '<blah blah>/pkg' contains errors.
INFO: Elapsed time: 0.149s
``````````

**Comment by prestonvanloon** [comment]
Duplicate of #65. I think the documentation needs to be updated.

**Comment by prestonvanloon** [comment]
@chaitanya9186 I updated the docs to show that `with_grpc` is not needed. You should be unblocked by removing that property. Since the build definition is `grpc_` we can assume that grpc is inherently true. :)

**Comment by chaitanya9186** [comment]
And also, how can I share grpc_gateway_proto_library rule between grpc_gateway_binary and go_binary.

Basically, I have helloworld.proto as mentioned above. I would like to share it with both gateway code and service implementation code.

I can use protos field but it will be generating code twice. Is there anyway I can generate code once.

**Comment by chaitanya9186** [comment]
Friendly ping. Thanks for the help in advance.

**Comment by prestonvanloon** [comment]
I don't think you can use a `grpc_gateway_proto_library` in a `grpc_gateway_binary`. However, it seems that since `binary` rule is a superset of the the `library` rule then you _should_ be able to use the `library` in the `binary` and share this dependency across many binaries. 

I would recommend you creating a new issue as a feature request since this is not possible at this time.

**Comment by chaitanya9186** [comment]
Created new issue but I am not sure how to make it a feature request.

**Comment by pcj** [comment]
Please re-open if still active, thanks.
